eZ/Publish/Core/Persistence/Cache/UserHandler.php

```diff
@@ -170,19 +224,50 @@ public function loadRoles()
      */
     public function loadRoleAssignment($roleAssignmentId)
     {
-        $this->logger->logCall(__METHOD__, array('assignment' => $roleAssignmentId));
+        $cacheItem = $this->cache->getItem("ez-role-assignment-${roleAssignmentId}");
+        if ($cacheItem->isHit()) {
+            return $cacheItem->get();
+        }
 
-        return $this->persistenceHandler->userHandler()->loadRoleAssignment($roleAssignmentId);
+        $this->logger->logCall(__METHOD__, array('assignment' => $roleAssignmentId));
+        $roleAssignment = $this->persistenceHandler->userHandler()->loadRoleAssignment($roleAssignmentId);
+
+        $cacheItem->set($roleAssignment);
+        $cacheItem->tag(
+            [
+                'role-assignment-group-list-' . $roleAssignment->contentId,
+                'role-assignment-role-list-' . $roleAssignment->roleId,
+                'role-assignment-' . $roleAssignmentId,
+            ]
+        );
+        $this->cache->save($cacheItem);
+
+        return $roleAssignment;
     }
 
     /**
      * {@inheritdoc}
      */
     public function loadRoleAssignmentsByRoleId($roleId)
     {
+        $cacheItem = $this->cache->getItem("ez-role-assignment-${roleId}-by-role");
+        if ($cacheItem->isHit()) {
+            return $cacheItem->get();
+        }
+
         $this->logger->logCall(__METHOD__, array('role' => $roleId));
+        $roleAssignments = $this->persistenceHandler->userHandler()->loadRoleAssignmentsByRoleId($roleId);
 
-        return $this->persistenceHandler->userHandler()->loadRoleAssignmentsByRoleId($roleId);
+        $cacheItem->set($roleAssignments);
+        $cacheTags = ['role-assignment-role-list-' . $roleId];
+        foreach ($roleAssignments as $roleAssignment) {
+            $cacheTags[] = 'role-assignment-' . $roleAssignment->id;
+            $cacheTags[] = 'role-assignment-group-list-' . $roleAssignment->contentId;
```

Comment:
You build `RoleAssignment` cache tags few times in this file, maybe consider`private function getCacheTags(RoleAssignment $roleAssignment, ...`